### PR TITLE
Update minimum required `frequenz-client-base` to 0.6.0

### PR DESCRIPTION
**Pull request by shsms**
This makes the weather client compatible with SDK 1.0.0-rc900.

This PR also updates this repo to use the latest repo-config v0.9.2

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (cwasicki) [inline .github/workflows/ci.yaml]: Your conflict resolution failed I think

Comment (shsms) [reply .github/workflows/ci.yaml]: Removed the update to repoconfig 0.10 for now.  So this PR takes it only upto 0.9.2

**Comment by llucax** [comment]
It looks like you also need to bump the `mkdocstrings` dependency.

Heads-up, the migration script for 0.10.0 fails often to update the dockerfile and `ci` workflow. It is very flaky when applying patches and screws everything up. It might be easier to update those manually or just ignore the updates for those files, as it is mostly (if not all) about supporting passing credentials when fetching dependencies, which is a non-issue for open source projects.

File: .github/workflows/ci.yaml
Comment (llucax) [reply .github/workflows/ci.yaml]: Oh, I guess you were already bitten by the flaky migration script.

**Review by llucax — APPROVED** [review]

**Comment by llucax** [comment]
I approved but needs approval from a code owner. I think since this is not touching the "business logic" you can just force-merge.

**Review by cwasicki — APPROVED** [review]

**Pull request merged**
